File: hackerrank/templates/template.lua
TEST = [[]]

function stringio(s)
  f = assert(io.tmpfile())
  f:write(s)
  f:flush()
  f:seek("set", 0)
  return f
end

function solve(input)
  -- test_cases = tonumber(input:read("*l"))
  -- for i=1, test_cases do
  --   line = input:read("*l")
  -- end
end

function main(...)
  if arg[1] == "test" then
    f = stringio(TEST)
    solve(f)
    f:close()
    return
  end

  solve(io.stdin)
end

main(...)

File: hackerrank/templates/template.rs
use std::io::{BufReader, BufRead};
use std::env;

#[allow(dead_code)]
static TEST: &'static str = "";

fn solve() -> {

}

fn setup(input: &mut BufRead) {
  let mut line = String::new();
  input.read_line(&mut line).ok().unwrap();
  let test_cases = line.trim().parse().unwrap();

  for _ in 0..test_cases {
    // line.clear();
    // input.read_line(&mut line).ok().unwrap();
    // let n = line.trim().parse().unwrap();
    // let whatever: Vec<usize> = line.split(' ').map(|x| x.trim().parse().unwrap()).collect();
  }
}

fn main() {
  // for testing purposes
  if let Some(arg) = env::args().nth(1) {
    if arg == "test" {
      let mut br = BufReader::new(TEST.as_bytes());
      setup(&mut br);
      return;
    }
  }

  let stdin = std::io::stdin();
  let mut br = BufReader::new(stdin);
  setup(&mut br);
}
